django_webapps/scripts/db_population/parsers/common_parsers.py

```diff
@@ -88,11 +104,110 @@ def parse_telegram_name(tg_str: str) -> str | None:
     if match_result is not None and len(match_result.group(0)) == len(tg_str):
         return f"@{match_result.group(0)}"
 
-    raise ValueError(f"tg: {tg_str} is not valid")
+    return None
+
+
+def find_any_phone_number(number_str: str) -> list[str]:
+    number_str = number_str.replace(" ", "")
+    regex = re.compile(r"[\+]?[(]?[0-9]{3}[)]?[-\s\.]?[0-9]{3}[-\s\.]?[0-9]{4,6}")
+    return re.findall(regex, number_str)
+
+
+def parse_phone_number(number_str: str) -> str | None:
+    number_str = number_str.strip().rstrip()
+    country_codes_to_test = ["", "+380", "+7", "+1"]
+
+    phone_numbers = find_any_phone_number(number_str)
+    phone_numbers.append(number_str)
+
+    parsed_number = None
+    for raw_phone_number in phone_numbers:
+        if parsed_number is not None:
+            break
+        for country_code in country_codes_to_test:
+            number_to_parse = country_code + raw_phone_number
+            try:
+                parsed_number = phonenumbers.parse(number_to_parse)
+                break
+            except phonenumbers.NumberParseException:
+                pass
+
+    if parsed_number is None:
+        raise ValueError(f"Unable to parse phone number: {number_str}")
+
+    return phonenumbers.format_number(parsed_number, phonenumbers.PhoneNumberFormat.E164)
```

Comment:
I think `phonenumbers` should have something like `.validate` too. A number could be parsed but be invalid. Like +1 314 123.

Reply:
The validation is working only if we provide a country code but we can not do it in advance

Reply:
IIRC you can set a country, and then `phonenumbers` will try and validate the number as a regional phone number for that country. We can try several most common countries: Ukraine, Russia, UK, Ireland, US, Germany, France, Spain

Reply:
Yes, I did that already for all coutnries

Reply:
I would still think that at least some note in comment field (phone number passed or failed `phonenumbers`' validation) would be very useful for the coordinators/admins.